### ISSUE TITLE
Migrate to v4 guidance is not good enough

### ISSUE THREAD

**Issue by zhuxb711** [jaredwray/keyv#1211]
**Describe the bug**
I'm using latest v3 keyv/redis currently. There is no enough information about how to migrate from v3 to v4 if using cluster replica set within k8s. In v4, using createCluster and it requests 'rootNodes' but in k8s we don't need to provide the node IP address, just provide a host URL. Would you please let me know how to migrate to v4?

Here is the code about how I make it work in v3:
```typescript
new KeyvRedis(new Redis.Cluster(
            [
              {
                host: configService.getOrThrow<string>('REDISDB_HOST'),
                port: Number(configService.getOrThrow<string>('REDISDB_PORT'))
              }
            ],
            {
              scaleReads: configService.getOrThrow<NodeRole>('REDISDB_READ_PREFERENCE'),
              enableReadyCheck: true,
              enableOfflineQueue: true,
              redisOptions: {
                autoResubscribe: true,
                autoResendUnfulfilledCommands: true,
                socketTimeout: ms(configService.getOrThrow<string>('REDISDB_SOCKET_TIMEOUT')),
                connectTimeout: ms(configService.getOrThrow<string>('REDISDB_CONNECT_TIMEOUT')),
                maxRetriesPerRequest: Number(configService.getOrThrow<string>('REDISDB_MAX_RETRY_ATTEMPTS_PER_REQUEST'))
              },
              clusterRetryStrategy: (times, reason) => {
                if (times <= Number(configService.getOrThrow<string>('REDISDB_MAX_CLUSTER_RETRY_ATTEMPTS'))) {
                  const nextDelay = Math.min(
                    100 * Math.pow(2, times),
                    ms(configService.get<string>('REDISDB_RETRIES_MAX_BACKOFF_LIMIT'))
                  );
                  return nextDelay;
                }

                throw new Error(`Unable to connect to the redis cluster and throttler guard could not be setup`);
              }
            }
          ))
```

**Comment by jaredwray** [comment]
@zhuxb711 - thanks for sending in the issue. The new redis driver is based on node-redis which you can follow their documentation on `createCluster`.

You can read more on this here and post any issues that you encounter in that repo. https://github.com/redis/node-redis/blob/master/docs/clustering.md

I will add in the link to this documentation in ours to help.

**Comment by jaredwray** [comment]
https://github.com/jaredwray/keyv/pull/1212

**Comment by zhuxb711** [comment]
I got type not assignable error from v4.0.1
```typescript
        const cluster = createCluster({
          rootNodes: [
            {
              url: `redis://host:port`
            }
          ],
          useReplicas: true
        });
        
        const key = new Keyv({
          store: new KeyvRedis(cluster) // cluster is not assignable to the constructor of 'KeyvRedis'
        });
```

**Comment by jaredwray** [comment]
@zhuxb711 the type issue should be resolved in `v4.0.2` please note that when using clustering `clear()` and `iterator()` do not work due to the lack of SCAN in clustering.
